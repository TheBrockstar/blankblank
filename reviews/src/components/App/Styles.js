import  styled  from 'styled-components';
import { flexContainer } from '../../globalStyles/Styles';

export const StyledHeader = styled.header `
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  background-color: rgb(81, 197, 212);
`

export const HeaderWrapper = styled(flexContainer)` 
height: 50px;
width: 80%;

`

export const StyledFooter = styled.footer `
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  background-color: rgb(234, 168, 58);
`

export const FooterWrapper = styled(flexContainer)`
height: 50px;
width: 80%;
max-width: 1080px;
`

export const ViewWrapper = styled(flexContainer)`
width: 100%;
height: auto;
min-height: calc(100vh - 100px);
`