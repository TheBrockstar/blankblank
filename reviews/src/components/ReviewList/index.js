import React, { useState, useEffect } from 'react';
import axios from 'axios';

import List from './List';

function ReviewList() {

  return (
    <section>
      <List />
    </section>
  )
}

export default ReviewList;

