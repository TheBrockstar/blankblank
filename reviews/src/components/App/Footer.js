import React from 'react';
import { StyledFooter, FooterWrapper } from './Styles';

const Footer = _ => {
  return (
    <FooterWrapper>
      <StyledFooter>
        This is a footer.
      </StyledFooter>
    </FooterWrapper>
  )
}

export default Footer;