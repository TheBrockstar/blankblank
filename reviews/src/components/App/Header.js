import React from 'react';
import { StyledHeader, HeaderWrapper } from './Styles';

const Header = _ => {
  return (
    <HeaderWrapper>
      <StyledHeader>
        <h1>Shakespeare Reviews</h1>
      </StyledHeader>
    </HeaderWrapper>
  )
}

export default Header;