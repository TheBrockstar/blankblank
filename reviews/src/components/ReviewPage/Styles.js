import styled from 'styled-components';
import { flexContainer } from '../../globalStyles/Styles';


// Review Page Components
export const ReviewContainer = styled.section`
  margin-top: 30px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
`;

export const Rating = styled(flexContainer)`
  width: 200px;
  height: 200px;
  flex-direction: column;
  background-color: whitesmoke;
  border-radius: .7rem;
  background-color: rgba(53, 23, 6, 0.5);
  color: rgb(81,197,212);;
  font-weight: bold;
  align-self: center;

  span {
    font-size: 5.2rem;
  }

  @media only screen and (max-width: 650px) {
    width: 160px;
    height: 160px;

    span {
      font-size: 4.2rem;
    }
  }

  @media only screen and (max-width: 450px) {
    width: 140px;
    height: 140px;

    span {
      font-size: 3.2rem;
    }
  }

`;

export const Author = styled.h1`
  font-size: 2.5rem;
  font-weight: bold;
  margin-top: 30px;
  align-self: center;
  text-decoration: underline;
  
  @media only screen and (max-width: 650px) {
    font-size: 2rem;
    margin-top: 30px;
  }

  @media only screen and (max-width: 450px) {
    font-size: 1.5rem;
    margin-top: 15px;
  }
`

export const Quote = styled.blockquote`
  max-width: 600px;
  font-style: italic;
  font-size: 2.5rem;
  padding-left: 30px;
  border-left: 10px solid rgb(105,68,37);
  margin: 30px 0;

  @media only screen and (max-width: 650px) {
    font-size: 2rem;
    margin: 20px 10px;
  }

  @media only screen and (max-width: 450px) {
    font-size: 1.5rem;
    margin: 10px 10px;
  }
`
export const ReviewDate = styled.p`
  font-size: 1.5rem;
  font-weight: bold;
  margin-top: 30px;

  @media only screen and (max-width: 650px) {
    font-size: 1rem;
    margin: 20px 10px;
  }

  @media only screen and (max-width: 450px) {
    font-size: .75rem;
    margin: 10px 10px;
  }
`

export const Stars = styled.div`
  font-size: 1.5rem;
  white-space: nowrap;
  overflow: hidden;
  width: ${props => ((props.rating / 5) * 100) + "%"} // Calculates the number of stars to show.

  @media only screen and (max-width: 650px) {
    font-size: 1.25rem;
  }

  @media only screen and (max-width: 450px) {
    font-size: 1rem;
  }
`;

export const ReturnToList = styled.button`
  align-self: center;
  background-color: rgba(254,197,107,0.3);
  box-shadow: 0 .1rem .2rem .1rem rgba(126,56,12,.5);
  margin-top: 60px;
  padding: 10px 15px;
  border: 0px;
  border-radius: 10px;
  font-size: 1.5rem;
  color: rgb(53, 23, 6);
  cursor: pointer;

  &:hover {
    background-color: rgba(254,197,107,0.5);
  }

  @media only screen and (max-width: 650px) {
    margin-top: 30px;
  }

  @media only screen and (max-width: 450px) {
    margin-top: 15px;
  }
`