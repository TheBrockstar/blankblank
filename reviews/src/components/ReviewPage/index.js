import React, { useState, useEffect } from 'react';
import axios from 'axios';

import { ReviewContainer, Rating, Stars, Quote, Author, ReviewDate, ReturnToList } from './Styles';

import { Loading, NotFound404 } from '../../globalStyles/Styles';

function ReviewPage(props) {

  const [review, setReview] = useState(null);

  useEffect(() => {
  async function fetchReview() {
    try {
      const response = await axios(
      `https://shakespeare.podium.com/api/reviews/${props.match.params.id}`, 
      { 'headers': { 'x-api-key':'H3TM28wjL8R4#HTnqk?c' }}
      );

      setReview(response.data);
    } catch ( error ) {
      setReview('404')
    }
  }

  fetchReview();

 ;
  }, []);
    if (review) {
      return review === "404" ? (
        <NotFound404>
          404 - Not Found 
          <br/> 
          Get Thee Hence!
        </NotFound404>
      ) : (
        <ReviewContainer>
          <Rating>
            <span>{ review.rating }</span>
            <div>
              <Stars rating={ review.rating }>★ ★ ★ ★ ★</Stars>
            </div>
          </Rating>
          <Author>{ review.author }</Author>
          <Quote>“{ review.body }”</Quote>
          <ReviewDate>Reviewed On: { (new Date(review.publish_date)).toDateString() }</ReviewDate>
          <ReturnToList role='link' onClick={_ => props.history.push('/')}>Return to Full List</ReturnToList>
        </ReviewContainer>
      );
    } else {
      return (<Loading>Loading...</Loading>)
    }
    
}

export default ReviewPage;
