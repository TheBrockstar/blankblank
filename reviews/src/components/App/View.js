import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { ViewWrapper } from './Styles';
import ReviewList from '../ReviewList';

const View = _ => {
  return (
    <main>
      <ViewWrapper>
          <Router>
            <Route exact path="/" component={ReviewList}/>
            <Route path="/other" render={_ => (<h1>Other Content</h1>)}/>
          </Router>
      </ViewWrapper>
    </main>
  )
}

export default View;