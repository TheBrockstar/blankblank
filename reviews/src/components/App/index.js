import React, { Component } from 'react';

import './App.css';

import Header from './Header';
import Footer from './Footer';
import View from './View';

class App extends Component {
  render() {
    return (
      <>
        <Header />
        <View />
        <Footer />
      </>
    );
  }
}

export default App;
