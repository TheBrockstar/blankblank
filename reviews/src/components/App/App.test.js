import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, unmount } from 'enzyme';
import { MemoryRouter, Route, Switch } from 'react-router';
import App from './index';
import View from './View';
import Footer from './Footer';
import Header from './Header';
import { NotFound404 } from '../../globalStyles/Styles';
import ReviewList from '../ReviewList';
import ReviewPage from '../ReviewPage';

describe ('App', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });

});


describe ('View', () => {
  it('matches snapshot', () => {
    expect(shallow(<View />)).toMatchSnapshot();
  });
});

describe ('Header', () => {
  it('matches snapshot', () => {
    expect(shallow(<Header />)).toMatchSnapshot();
  });
});

describe ('Footer', () => {
  it('matches snapshot', () => {
    expect(shallow(<Footer />)).toMatchSnapshot();
  });
});

describe ('Routes', () => {

  it('Render `ReviewList` if path is `/`', () => {

    const wrapper = mount(
      <MemoryRouter initialEntries={[{pathname:"/"}]} initialIndex={0}>
        <View />
      </MemoryRouter>
    );

    expect(
      wrapper.find(ReviewPage)
    ).toHaveLength(0);
    expect(
      wrapper.find(NotFound404)
    ).toHaveLength(0);
    expect(
      wrapper.find(ReviewList)
    ).toHaveLength(1);

    
  wrapper.unmount();
  });

  it('Render `NotFound404` if path is invalid.', () => {

    const wrapper = mount(
      <MemoryRouter initialEntries={[{pathname:"/incorrect"}]} initialIndex={0}>
        <Switch>
          <Route exact path="/" render={props => <ReviewList { ...props } />}/>
          <Route path="/review/:id" render={props => (<ReviewPage  { ...props }/>)}/>
          <Route path="/" render={_ => 
            (<NotFound404>404 - Not Found <br/> Get Thee Hence!</NotFound404>)}
            />
        </Switch>
      </MemoryRouter>
    );

    expect(
      wrapper.find(ReviewPage)
    ).toHaveLength(0);
    expect(
      wrapper.find(NotFound404)
    ).toHaveLength(1);
    expect(
      wrapper.find(ReviewList)
    ).toHaveLength(0);

    
  wrapper.unmount();
  });

  it('Render `ReviewPage` if path matches `/review/:id`', () => {

    const wrapper = mount(
      <MemoryRouter initialEntries={[{pathname:"/review/12345"}]} initialIndex={0}>
        <Switch>
          <Route exact path="/" render={props => <ReviewList { ...props } />}/>
          <Route path="/review/:id" render={props => (<ReviewPage  { ...props }/>)}/>
          <Route path="/" render={_ => 
            (<NotFound404>404 - Not Found <br/> Get Thee Hence!</NotFound404>)}
            />
        </Switch>
      </MemoryRouter>
    );

    expect(
      wrapper.find(ReviewPage)
    ).toHaveLength(1);
    expect(
      wrapper.find(NotFound404)
    ).toHaveLength(0);
    expect(
      wrapper.find(ReviewList)
    ).toHaveLength(0);

    
  wrapper.unmount();
  });


});
