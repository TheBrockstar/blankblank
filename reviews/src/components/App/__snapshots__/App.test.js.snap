// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header />,
        <View />,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header />,
          <View />,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Footer matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <h1>
          © 1616 - The Bard of Avon
        </h1>
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <h1>
          © 1616 - The Bard of Avon
        </h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "© 1616 - The Bard of Avon",
        },
        "ref": null,
        "rendered": "© 1616 - The Bard of Avon",
        "type": "h1",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bZQynM",
          "isStatic": true,
          "lastClassName": "GWuRD",
          "rules": Array [
            "
  display: flex;
  justify-content: center;
  align-items: center;
",
            "
height: 60px;
width: 80%;
max-width: 1080px;

@media only screen and (max-width: 850px) {
  width: 96%;
}
",
          ],
        },
        "displayName": "Styled(styled.div)",
        "foldedComponentIds": Array [
          "sc-bdVaJa",
        ],
        "render": [Function],
        "styledComponentId": "sc-bZQynM",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": true,
        "lastClassName": "hKCajt",
        "rules": Array [
          "
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  background-color: rgb(81, 197, 212);
  border-top: 5px solid rgb(105, 68, 37);
  margin-top: 15px;

  h1 {
    font-size: 1.2rem;
    margin-top: 5px;
    color: whitesmoke;
  }
",
        ],
      },
      "displayName": "styled.footer",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "footer",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <h1>
            © 1616 - The Bard of Avon
          </h1>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h1>
            © 1616 - The Bard of Avon
          </h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "© 1616 - The Bard of Avon",
          },
          "ref": null,
          "rendered": "© 1616 - The Bard of Avon",
          "type": "h1",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": true,
            "lastClassName": "GWuRD",
            "rules": Array [
              "
  display: flex;
  justify-content: center;
  align-items: center;
",
              "
height: 60px;
width: 80%;
max-width: 1080px;

@media only screen and (max-width: 850px) {
  width: 96%;
}
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "foldedComponentIds": Array [
            "sc-bdVaJa",
          ],
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-EHOje",
          "isStatic": true,
          "lastClassName": "hKCajt",
          "rules": Array [
            "
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  background-color: rgb(81, 197, 212);
  border-top: 5px solid rgb(105, 68, 37);
  margin-top: 15px;

  h1 {
    font-size: 1.2rem;
    margin-top: 5px;
    color: whitesmoke;
  }
",
          ],
        },
        "displayName": "styled.footer",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-EHOje",
        "target": "footer",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Header matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <h1>
          - Shakespeare Reviews -
        </h1>
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <h1>
          - Shakespeare Reviews -
        </h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "- Shakespeare Reviews -",
        },
        "ref": null,
        "rendered": "- Shakespeare Reviews -",
        "type": "h1",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-ifAKCX",
          "isStatic": true,
          "lastClassName": "eJNCqI",
          "rules": Array [
            "
  display: flex;
  justify-content: center;
  align-items: center;
",
            " 
height: 60px;
width: 80%;

@media only screen and (max-width: 850px) {
  width: 96%;
}
",
          ],
        },
        "displayName": "Styled(styled.div)",
        "foldedComponentIds": Array [
          "sc-bdVaJa",
        ],
        "render": [Function],
        "styledComponentId": "sc-ifAKCX",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": true,
        "lastClassName": "gRyEjJ",
        "rules": Array [
          "
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  background-color: rgb(81, 197, 212);
  border-bottom: 5px solid rgb(105, 68, 37);
  margin-bottom: 15px;

  h1 {
    font-size: 2.5rem;
    color: whitesmoke
    margin-top: 5px;
  }

  @media only screen and (max-width: 850px) {
    h1 {
    font-size: 2rem;
    }
  }

  @media only screen and (max-width: 450px) {
    h1 {
    font-size: 1.7rem;
    }
  }
",
        ],
      },
      "displayName": "styled.header",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "header",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <h1>
            - Shakespeare Reviews -
          </h1>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h1>
            - Shakespeare Reviews -
          </h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "- Shakespeare Reviews -",
          },
          "ref": null,
          "rendered": "- Shakespeare Reviews -",
          "type": "h1",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "lastClassName": "eJNCqI",
            "rules": Array [
              "
  display: flex;
  justify-content: center;
  align-items: center;
",
              " 
height: 60px;
width: 80%;

@media only screen and (max-width: 850px) {
  width: 96%;
}
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "foldedComponentIds": Array [
            "sc-bdVaJa",
          ],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "lastClassName": "gRyEjJ",
          "rules": Array [
            "
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  background-color: rgb(81, 197, 212);
  border-bottom: 5px solid rgb(105, 68, 37);
  margin-bottom: 15px;

  h1 {
    font-size: 2.5rem;
    color: whitesmoke
    margin-top: 5px;
  }

  @media only screen and (max-width: 850px) {
    h1 {
    font-size: 2rem;
    }
  }

  @media only screen and (max-width: 450px) {
    h1 {
    font-size: 1.7rem;
    }
  }
",
          ],
        },
        "displayName": "styled.header",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "header",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`View matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ForwardRef>
        <BrowserRouter>
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              path="/review/:id"
              render={[Function]}
            />
            <Route
              path="/"
              render={[Function]}
            />
          </Switch>
        </BrowserRouter>
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <BrowserRouter>
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              path="/review/:id"
              render={[Function]}
            />
            <Route
              path="/"
              render={[Function]}
            />
          </Switch>
        </BrowserRouter>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              path="/review/:id"
              render={[Function]}
            />
            <Route
              path="/"
              render={[Function]}
            />
          </Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                path="/review/:id"
                render={[Function]}
              />,
              <Route
                path="/"
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/review/:id",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-gzVnrw",
          "isStatic": true,
          "lastClassName": "fmPKvS",
          "rules": Array [
            "
  display: flex;
  justify-content: center;
  align-items: center;
",
            "
width: 100%;
height: auto;
align-items: flex-start;
min-height: calc(100vh - 160px);
",
          ],
        },
        "displayName": "Styled(styled.div)",
        "foldedComponentIds": Array [
          "sc-bdVaJa",
        ],
        "render": [Function],
        "styledComponentId": "sc-gzVnrw",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ForwardRef>
          <BrowserRouter>
            <Switch>
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />
              <Route
                path="/review/:id"
                render={[Function]}
              />
              <Route
                path="/"
                render={[Function]}
              />
            </Switch>
          </BrowserRouter>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <BrowserRouter>
            <Switch>
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />
              <Route
                path="/review/:id"
                render={[Function]}
              />
              <Route
                path="/"
                render={[Function]}
              />
            </Switch>
          </BrowserRouter>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Switch>
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />
              <Route
                path="/review/:id"
                render={[Function]}
              />
              <Route
                path="/"
                render={[Function]}
              />
            </Switch>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  exact={true}
                  path="/"
                  render={[Function]}
                />,
                <Route
                  path="/review/:id"
                  render={[Function]}
                />,
                <Route
                  path="/"
                  render={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/review/:id",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": true,
            "lastClassName": "fmPKvS",
            "rules": Array [
              "
  display: flex;
  justify-content: center;
  align-items: center;
",
              "
width: 100%;
height: auto;
align-items: flex-start;
min-height: calc(100vh - 160px);
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "foldedComponentIds": Array [
            "sc-bdVaJa",
          ],
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
